src/intersections/triangles.jl

```diff
@@ -188,3 +188,20 @@ function intersecttype(f::Function, r::Ray{3,T}, t::Triangle{3,T}) where {T}
 
   return IntersectingRayTriangle(r(λ)) |> f
 end
+
+"""
+    intersecttype(s, p)
+
+Calculate the intersection of a ray and polygon in 3D
+and apply function `f` to it.
+"""
+function intersecttype(f::Function, r::Ray{3,T}, p::Ngon{N,3,T}) where {N,T}
+  for t in collect(triangulate(p))
```

Comment:
Do we really need to collect the resulting mesh?

Reply:
@juliohm I currently have no better approach.

Reply:
What I mean is that `Mesh` types are iterable, you can omit the `collect` call. `for t in triangulate(p)` should work and it will materialize one triangle at a time.

Reply:
@juliohm Thank you for your suggestion. Now I have been changed.